packages/core/src/model.js

```diff
@@ -3188,9 +3188,16 @@ Instead of specifying a Model, either:
       || options.clone
     ) {
       const values = Object.create(null);
+      const modelDefinition = this.constructor.modelDefinition;
       if (attributesWithGetters.size > 0) {
         for (const attributeName2 of attributesWithGetters) {
-          if (!this._options.attributes?.includes(attributeName2)) {
+          const attributeDefinition = modelDefinition.rawAttributes[attributeName2];
+          // attributeDefinition is undefined , a virtual getter directly invoke it
+          // if attributeName2 is absent in this.options.attributes and
+          // if attributeName2 is not a virtual attribute, do not call its getter
+          if (!(attributeDefinition === undefined)
+            && (!this._options.attributes?.includes(attributeName2)
+            && !Object.prototype.hasOwnProperty.call(attributeDefinition.type, DataTypes.VIRTUAL))) {
```

Comment:
This line is always going to be false, that's not how you check if something is an instance of VIRTUAL

But even then, all getters must be called, not just the ones on virtual attributes

Reply:
Sorry, I missed the attribute and attributeWithGetters instance and I agree it is redundant.

Also I ran the added test on sequelize : main branch  and it is also passing the test. So issue #11041 should not be an issue for main branch.

This pr can be closed.

Reply:
Maybe only keep the new test in this PR? So we don't accidentally introduce a regression later